src/main/java/io/temporal/internal/replay/ReplayWorkflowExecutor.java

```diff
@@ -395,96 +242,48 @@ private void handleWorkflowExecutionSignaled(HistoryEvent event) {
   }
 
   @Override
-  public WorkflowTaskResult handleWorkflowTask(PollWorkflowTaskQueueResponseOrBuilder workflowTask)
-      throws Throwable {
+  public WorkflowTaskResult handleWorkflowTask(
+      PollWorkflowTaskQueueResponseOrBuilder workflowTask) {
     lock.lock();
     try {
+      long startTime = System.currentTimeMillis();
       queryResults.clear();
-      boolean forceCreateNewWorkflowTask = handleWorkflowTaskImpl(workflowTask, null);
-      return new WorkflowTaskResult(
-          commandHelper.getCommands(), queryResults, forceCreateNewWorkflowTask, completed);
+      handleWorkflowTaskImpl(workflowTask);
```

Comment:
[Nit] kinda strange that some of these methods communicate their results via return values while others communicate using instance members of the class.

Reply:
It would be a pretty large change. So far public methods return values and private ones communicate through fields.